pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -130,6 +136,28 @@ func (r *pullthroughBlobStore) ServeBlob(ctx context.Context, w http.ResponseWri
 	return nil
 }
 
+// Get attempts to fetch the requested blob by digest using a remote proxy store if necessary.
+func (r *pullthroughBlobStore) Get(ctx context.Context, dgst digest.Digest) ([]byte, error) {
+	store, ok := r.digestToStore[dgst.String()]
+	if !ok {
+		data, getErr := r.BlobStore.Get(ctx, dgst)
+		if getErr == nil {
+			return data, nil
+		}
+
+		desc, err := r.remoteStat(ctx, dgst)
+		if err != nil {
+			return nil, getErr
+		}
+		store, ok = r.digestToStore[desc.Digest.String()]
+		if !ok {
+			return nil, getErr
```

Comment:
> return something like fmt.Errorf("unable to find digest foo in store")

I'd rather return `getErr` again because it returnes expected errors to the caller which interprets them a transforms to api error codes.

I can log the error though.